opensrp-reveal/src/main/java/org/smartregister/reveal/presenter/RevealJsonFormFragmentPresenter.java

```diff
@@ -31,11 +45,75 @@ public ValidationStatus writeValuesAndValidate(LinearLayout mainView) {
                 if (childAt instanceof RevealMapView) {
                     RevealMapView mapView = (RevealMapView) childAt;
                     validationStatus = GeoWidgetFactory.validate(formFragment, mapView);
+                    if (validationStatus.isValid()) {
+                        if (BuildConfig.VALIDATE_FAR_STRUCTURES) {
+                            validateUserLocation(formFragment.getActivity(), mapView);
+                        } else {
+                            onValidateUserLocation(true);
+                        }
+                    }
                 }
             }
 
         }
         return validationStatus;
     }
 
+
+    private void validateUserLocation(Activity activity, RevealMapView mapView) {
+        if (ActivityCompat.checkSelfPermission(activity,
+                Manifest.permission.ACCESS_FINE_LOCATION) == PackageManager.PERMISSION_GRANTED) {
+            mapView.getFusedLocationClient().getLastLocation()
+                    .addOnSuccessListener(location -> {
+                        if (location != null) {
+                            LatLng mapPosition = mapView.getMapboxMap().getCameraPosition().target;
```

Comment:
Is this a placeholder for the index case structure coordinates?

If so, maybe add a method like `getIndexCaseCoordinates` that for now can return `mapView.getMapboxMap().getCameraPosition().target` with its specifics to be implemented later?

Reply:
No this validating adding new points

`mapView.getMapboxMap().getCameraPosition().target` would be the point to be added and needs to be validated if user geographically near that point

Reply:
Cool.